src/System.Private.ServiceModel/src/SMDiagnostics/System/ServiceModel/Diagnostics/ExceptionUtility.cs

```diff
@@ -85,16 +87,39 @@ public Exception ThrowHelperCallback(Exception innerException)
 
         public Exception ThrowHelperCritical(Exception exception)
         {
-            return exception;
+            return ThrowHelper(exception, EventLevel.Critical);
         }
 
         public Exception ThrowHelperError(Exception exception)
         {
-            return exception;
+            return ThrowHelper(exception, EventLevel.Error);
         }
 
         public Exception ThrowHelperWarning(Exception exception)
         {
+            return ThrowHelper(exception, EventLevel.Warning);
+        }
+
+        internal Exception ThrowHelper(Exception exception, EventLevel eventLevel)
+        {
+            IDictionary data = exception.Data;
+            if (data != null && !data.IsReadOnly && !data.IsFixedSize)
```

Comment:
Is adding extra info to the exception data doing anything for us? Does it end up in the final etw event? I have a suspicion this only had an effect when using tracesource.